tools/changelog-gen-for-net/Azure.SDK.ChangelogGen/Compare/Change.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.SDK.ChangelogGen.Compare
+{
+    public enum ChangeCatogory
+    {
+        Added,
+        Removed,
+        Obsoleted,
+        Updated,
+    }
+
+    public enum ChangeTarget
+    {
+        Type,
+        Constructor,
+        Method,
+        Property,
+        PropertyMethod,
+        StringValue,
+    }
+
+    public abstract class Change
+    {
+        public ChangeCatogory ChangeCatogory { get; init; }
+        public ChangeTarget Target { get; init; }
+
+        public Change(ChangeTarget target, ChangeCatogory category)
+        {
+            this.ChangeCatogory = category;
+            this.Target = target;
+        }
+
+        public abstract string Description { get; }
+        public virtual bool IsBreakingChange
```

Comment:
The other examples are also breaking changes and will need to be documented and treated that way.   As mentioned, my fear is that people will rely on this generator to identify breaking changes for them and miss those that are not "we removed a thing."

Reply:
the breaking changes from other examples will also be detected by the tool because the original interface no longer exists after the change. For example, if Method(string a, int b) is changed to Method(int b, string a). the tool will find the Method (string a, int b) can't be found in new version (removed) and log a breaking change for it. 
If we think a breaking change in a general way, it's just because an interface no longer available in the new version (treated as removed by the tool) no matter it's because the interface is deleted or renamed or modified to a different signature...

Reply:
That's fair.  I overlooked that the signature comparison would catch the two structural cases:

- The order of parameters was changed
- A parameter was renamed

The other two cases would require semantic analysis to catch.  Since we're scoping this to just management libraries and Arthur already signed-off, I won't block on this but I do still want to register my concerns about it - particularly for partner teams.